### ISSUE TITLE
Unknown compiler flag: --3str

### ISSUE THREAD

**Issue by Snell1224** [ovis-hpc/ovis#409]
I’ve been getting the following error during installation of the most recent OVIS-4 code:

make[3]: Leaving directory `/home/spwalto/ovis-9/22/LDMS_install/ldms/python/cmds'
make[3]: Entering directory `/home/spwalto/ovis-9/22/LDMS_install/ldms/python'
echo PYTHON_LDFLAGS are "-L/usr/lib64 -lpython3.6m"
PYTHON_LDFLAGS are -L/usr/lib64 -lpython3.6m
cython --fast-fail --gdb --3str -I ../../../ldms/python ../../../ldms/python/ldms.pyx -o ldms_pyx.c
Unknown compiler flag: --3str
make[3]: *** [ldms_pyx.c] Error 1
make[3]: Leaving directory `/home/spwalto/ovis-9/22/LDMS_install/ldms/python'
make[2]: *** [install-recursive] Error 1

Steps I’ve taken:
•	Cloned the ovis repository
•	Switched to OVIS-4 branch 
•	Ran the installation instructions from the github repo (error showed up during make and make install). 

The installation works with an older branch in my forked repo so this might have something to do with the recent Python3 migration.

- OS Name and Version: 
NAME="Red Hat Enterprise Linux Server"
VERSION="7.8 (Maipo)"
ID="rhel"

- Cython version 0.19

- Python 2.7.5

**Comment by nick-enoent** [comment]
The latest OVIS-4 no longer supports python2. If you'd like to use the latest OVIS-4, you'll need to upgrade your system to Python3.

**Comment by tom95858** [comment]
@Snell1224 that Makefile.am is old. The top of tree has this:

```
$(LDMS_PYX_C): ldms.pyx ldms.pxd
	echo PYTHON_LDFLAGS are "$(PYTHON_LDFLAGS)"
	cython -3 --directive language_level=3 --fast-fail --gdb -I $(srcdir) $< -o $@
```

Also, the Cython installed on your build machine is old. You'll need to do this:

```
pip install --upgrade cython
```

**Comment by baallan** [comment]
We have many pure-python scripts not dependent on cython or the cython-wrapped libraries.

We have the option of making them (pure python) unconditionally included (since python 3.6 appears to be python3 in base rhel and suse15).

If being purist (which in this case I would favor), we should have --enable-python-api (default disable) and --disable-python-scripts (default enable); I could be persuaded that they should both be default disabled. The need for cython should definitely be default disabled.
